pyramid_transactional_celery/transactional_task.py

```diff
@@ -85,19 +88,15 @@ def tpc_abort(self, transaction):
 
 
 class TransactionalTask(Task):
-    """A task whose execution is delayed until after the current transaction.
-    """
-    abstract = True
-
-    def original_apply_async(self, *args, **kwargs):
-        """Shortcut method to reach real implementation of
-        celery.Task.apply_sync"""
-        return super(TransactionalTask, self).apply_async(*args, **kwargs)
+    """A task whose execution is delayed until the current transaction gets committed."""
 
     def apply_async(self, *args, **kwargs):
-        if self.app.conf.get('CELERY_ALWAYS_EAGER', False):
-            return super(TransactionalTask, self).apply_async(*args, **kwargs)
-        else:
+        retries_count = kwargs.get('retries', 0)
+        call_from_tpc_finish = kwargs.get('call_from_tpc_finish', None)
+
+        if retries_count == 0 and not(call_from_tpc_finish):
             _get_manager().append((self, args, kwargs))
+        else:
+            return super(TransactionalTask, self).apply_async(*args, **kwargs)
```

Comment:
Este bloco podia ser simplificado para:
```suggestion
        if call_from_tpc_finish:
            return super(TransactionalTask, self).apply_async(*args, **kwargs)
        _get_manager().append((self, args, kwargs))
```

Reply:
heheh, isso é questão de opinião. Eu acho o else didático às vezes.

Reply:
É um warning no `pylint`... :grimacing: 
```bash
$ pylint foo.py
...
Unnecessary "else" after "return" (no-else-return)
```